test/tools/yulopti.cpp

```diff
@@ -87,37 +89,70 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	shared_ptr<Object> getSubObject(shared_ptr<Object> const& _object, string const& _qualifiedPath)
```

Comment:
Added to `Object` as a static function.

Not sure what kind of errors I'm supposed to throw. `Object::pathToSubObject()` uses only asserts. For now I've made it return null if the path isn't found as suggested in another comment.

Reply:
It uses asserts because the caller is expected to check for these things (they're basically preconditions). We have proper validations against them at other layers that are reported as compilation errors and once you get to actually using the `Object`, they should never happen.

I think it would be best to take the same approach in yulopti but if there is some specific case where it's problematic, we could raise an exception. Either one of those already available in `Exceptions.h` or a new one defined specifically for this.